drivers/usbhid-ups.c

```diff
@@ -2392,10 +2405,25 @@ static void ups_status_set(void)
 	if (ups_status & (STATUS(BYPASSAUTO) | STATUS(BYPASSMAN))) {
 		status_set("BYPASS");		/* on bypass */
 	}
+	if (ups_status & STATUS(ECOMODE)) {
+		status_set("ECO");		/* on ECO Mode */
+	}
 	if (ups_status & STATUS(OFF)) {
 		status_set("OFF");		/* ups is off */
 	}
 
+	/* Fix (OB+DISCHRG) for Eaton 9E Model */
+	if (!(ups_status & STATUS(ONLINE)) &&
+		(ups_status & STATUS(DISCHRG))) {
+		status_set("OB"); /* on battery */
+	}
+
+	/* Fix (OL+CHRG) for Eaton 9E Model */
+	if ((current_charge == onlinecharge_hovercharge) &&
+	   (ups_status & STATUS(ONLINE)) && (ups_status & STATUS(CHRG))) {
+		status_set("OL");
+	}
+
```

Comment:
Not sure about these recent additions, forcing `OB` on any status `!OL` + `DISCHRG` would set a device in e.g. `CAL DISCHRG` to `OB` as well and possibly raise notifications or even shutdowns (?). The second addition also, hard setting a `OL CHRG` to `OL` for all UPS using `usbhid-ups` (= multiple vendors and many devices) feels invasive to me and should be a user toggle if anything - in my opinion. Some users may want to know if their UPS is in fact (over-)charging despite being at 100%. I think we should obscure the actual UPS reported statuses as little as possible and really just when it's obvious it's a mistake or useless for all users, otherwise introduce configurational toggles for such things instead. Thoughts, @jimklimov ?

Reply:
I agree; the combo of "Fix <XXX> for <model>" without a toggle or model detection spoke for itself :\